parquet-avro/src/main/java/org/apache/parquet/avro/AvroRecordConverter.java

```diff
@@ -827,6 +832,14 @@ public void end() {
     }
   }
 
+  // Converter used to test whether a requested schema is a 2-level schema.
+  // This is used to convert the file's type assuming that the file uses
+  // 2-level lists and the result is checked to see if it matches the requested
+  // element type. This should always convert assuming 2-level lists because
+  // 2-level and 3-level can't be mixed.
+  private static final AvroSchemaConverter CONVERTER =
+      new AvroSchemaConverter(true);
+
```

Comment:
Instead of always set `AvroScehmaConverter.assumeRepeatedIsListElement` to true, shall we pass the properly constructed `AvroScehmaConverter` from `AvroReadSupport` to `AvroRecordMaterializer`, and then here?